pkg/apis/core/validation/validation.go

```diff
@@ -3735,12 +3773,26 @@ func ValidatePodUpdate(newPod, oldPod *core.Pod) field.ErrorList {
 		allErrs = append(allErrs, field.Invalid(specPath.Child("activeDeadlineSeconds"), newPod.Spec.ActiveDeadlineSeconds, "must not update from a positive integer to nil value"))
 	}
 
+	if utilfeature.DefaultFeatureGate.Enabled(features.InPlacePodVerticalScaling) {
+		// reject QOS change attempt
```

Comment:
Unrelated?

Reply:
legit. we will allow user to change resources but can't make a burstable class pod guaranteed and so on.